src/storage/src/hummock/iterator/merge_inner.rs

```diff
@@ -12,73 +12,75 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+use std::cmp::Ordering;
 use std::collections::binary_heap::PeekMut;
 use std::collections::{BinaryHeap, LinkedList};
 use std::sync::Arc;
 
 use async_trait::async_trait;
 use risingwave_hummock_sdk::VersionedComparator;
 
-use super::variants::*;
-use crate::hummock::iterator::{BoxedHummockIterator, HummockIterator};
+use crate::hummock::iterator::{
+    BoxedHummockIterator, DirectionEnum, HummockIterator, HummockIteratorDirection,
+};
 use crate::hummock::value::HummockValue;
 use crate::hummock::HummockResult;
 use crate::monitor::StateStoreMetrics;
 
-pub struct Node<'a, const DIRECTION: usize>(BoxedHummockIterator<'a>);
+pub struct Node<'a, D: HummockIteratorDirection, T: Send + Sync> {
+    iter: BoxedHummockIterator<'a, D>,
+    extra_info: T,
+}
 
-impl<const DIRECTION: usize> PartialEq for Node<'_, DIRECTION> {
+impl<T: Eq + Send + Sync, D: HummockIteratorDirection> PartialEq for Node<'_, D, T> {
```

Comment:
Do we really need to impl Eq for Node? I guess BinaryHeap doesn't rely on this.

Reply:
`Ord` relies on it.
```
pub trait Ord: Eq + PartialOrd<Self> {
...
}
```